### ISSUE TITLE
Delete Operation has an empty input.Document 

### ISSUE THREAD

**Issue by BrunoPereira** [rwynn/monstache#129]
@rwynn 

I'm using an highly plugin, using both the ES and Mongo clients and the bulk ES client.
No Problem here... works great.

But I've noticed that I had more documents in ES than I had in Mongo, so I Introduced a small debug log...
**Code:**
	if input.Operation == "d" {
		fmt.Printf("DEBUG - We are deleting the %v object:%v\n", input.Namespace, input.Document)
	}
**Log:**
DEBUG - We are deleting the STABLE.User object:map[]

What I'm reading from this is that the  input.Document is an empty map. Shouldn't be the object that I'm deleting?

PS:Since I'm doing a bunch of other things if I get the delete, the fact the document is empty might explain why the deletes aren't been applied.... 

Tnx

**Comment by rwynn** [comment]
@BrunoPereira I just pushed a release with a fix to put the `_id` field in the `input.Document` for deletes.

**Comment by rwynn** [comment]
@BrunoPereira sorry for this silly mistake.  I just pushed a new version that sets the map so that it is not nil.

**Comment by BrunoPereira** [comment]
@rwynn I'm the one that thanks you for all your hard work and support ....

So that fix took care of the nil map.
Now I have a different problem.... the delete is using the original Namespace as index instead of the one I'm given it....
 ```
func addRecordToBulkElastic(bulkElastic *elastic.BulkProcessor, typeName string, record map[string]interface{}) {
	req := elastic.NewBulkUpdateRequest()
	var recordID = record["_id"].(string)
	req.Index(index)
	req.Id(recordID)
	req.Type(typeName)
	req.DocAsUpsert(true)
	//DO NOT remove this. If you send IDs in the req body ES will discard it as malformed
	//The ID is send in the req.ID
	delete(record, "_id")
	delete(record, "_meta_monstache")
	req.Doc(record)

	if _, err := req.Source(); err == nil {
		bulkElastic.Add(req)
	} else {
		fmt.Printf("ERROR - The record with type:%s and id:%s is malformed, we won't send it to ES \n", typeName, recordID)
	}
	return
} 
```
This is the func I use to add all kind of operations to the Bulk Elastic Client.
The req.Index(index) uses a cons that I set at the start.
When I get  delete, although I set his Index, he's send to the ES with the "input.Namespace" as Index.
Can you check if the bug is on monstche or in the elastic.v5 lib ... or in my code above....

**Comment by rwynn** [comment]
For deletes I think you would want to use `NewBulkDeleteRequest`. 

https://github.com/olivere/elastic/blob/release-branch.v6/bulk_delete_request.go#L51

I'm not sure why it would be using the wrong index though.

**Comment by BrunoPereira** [comment]
@rwynn 
Tnx for the help.
I still need to test it with the alter config, but the deletes worked great in the ES index and now I have the same count in ES and Mongo

**Comment by rwynn** [comment]
This is now fixed as of `4.12.1` and `3.19.1`.
